Website/src/rendering/src/pages/api/play-shop-discover-to-ordercloud-feed-importer/updateordercloudcatalog.ts

```diff
@@ -0,0 +1,330 @@
+        await createSpecProductAssignment(productId, specName);
+      }
+      specListOrder++;
+    }
+  }
+}
+
+async function createSpecOption(
+  productId: string,
+  specName: string,
+  variantRow: VariantRow,
+  listOrder: number
+) {
+  await OrderCloudSDK.Specs.SaveOption(
+    `${productId}-${specName}`,
+    `${productId}-${specName}-${variantRow[specName.toLowerCase() as keyof typeof variantRow]}`,
+    {
+      ID: `${productId}-${specName}-${
+        variantRow[specName.toLowerCase() as keyof typeof variantRow]
+      }`,
+      Value: variantRow[specName.toLowerCase() as keyof typeof variantRow],
+      ListOrder: listOrder,
+    }
+  );
+}
+
+async function createSpecProductAssignment(productId: string, specName: string) {
+  await OrderCloudSDK.Specs.SaveProductAssignment({
+    SpecID: `${productId}-${specName}`,
+    ProductID: productId,
+  });
+}
+
+async function generateVariants(productIdToVariantRowsMap: Map<string, VariantRow[]>) {
+  const productIds = productIdToVariantRowsMap.keys();
+  for (const productId of productIds) {
+    // Generate the variants for this specific product
+    await OrderCloudSDK.Products.GenerateVariants(productId, { overwriteExisting: true });
+  }
+}
+
+async function updateVariants(productIdToVariantRowsMap: Map<string, VariantRow[]>) {
+  const productIds = productIdToVariantRowsMap.keys();
+  for (const productId of productIds) {
+    // Retrieve the variants for this specific product
+    let productVariants: OrderCloudSDK.Variant[] = [];
+    // Variants are returned in reverse order than the one that appears in the product feed
+    productVariants = (await OrderCloudSDK.Products.ListVariants(productId))?.Items;
+
+    // Update the variants (IDs, imageUrls) of this specific product
+    for (const variant of productVariants) {
+      const variantRow: VariantRow = productIdToVariantRowsMap.get(productId)?.pop();
+      const additionalSkuImageUrls: { Url: string; ThumbnailUrl: string }[] =
+        variantRow.additional_sku_image_urls.length > 0
+          ? variantRow.additional_sku_image_urls.split('|').map((imgUrl: string) => ({
+              Url: imgUrl,
+              ThumbnailUrl: '',
+            }))
+          : [];
+
+      await OrderCloudSDK.Products.PatchVariant(productId, variant.ID, {
+        ID: variantRow.sku,
+        xp: {
+          Images: [
+            {
+              Url: variantRow.sku_image_url,
+              ThumbnailUrl: '',
+            },
+            ...additionalSkuImageUrls,
+          ],
+          SkuUrl: variantRow.sku_url,
+        },
+      });
+    }
+  }
+}
+
+const handler: NextApiHandler<unknown> = async (_request, response) => {
+  try {
+    OrderCloudSDK.Configuration.Set({
+      baseApiUrl: process.env.NEXT_PUBLIC_ORDERCLOUD_BASE_API_URL,
+    });
+
+    const catalogList = await OrderCloudSDK.Catalogs.List();
+    const catalogID = catalogList.Items[0].ID;
+
+    await postCategories(catalogID);
+
+    await postProducts();
```

Comment:
Could we please split that call into a different route? /api/syncOrderCloudProducts